scripts/release-notes/index.js

```diff
@@ -0,0 +1,158 @@
+ *
+ * USAGE:
+ *  1) Make sure all the issues in the release are assigned to the milestone
+ *  2) Each issue should have one and only one Type label - the script will tell you which ones don't
+ *  3) Make sure you have generated a GH token and created a token.json file, eg: { "githubApiToken": "..." }
+ *     This token needs at least `read:org` permissions
+ *  4) Execute the command: node index.js <project_name> > <output_file>
+ *      eg: node index.js 3.0.0 > tmp.md
+ *  5) Insert the contents of the output file into the appropriate location in Changes.md
+ */
+
+const { Octokit } = require('@octokit/rest');
+const octokit = new Octokit({
+  auth: require('../token.json').githubApiToken,
+  userAgent: 'cht-release-note-generator',
+});
+
+const OWNER = 'medic';
+
+const REPO_NAME = process.argv[2];
+if (!REPO_NAME) {
+  throw new Error('You must specify a repo name (eg: "cht-core") as the first argument');
+}
+const MILESTONE_NAME = process.argv[3];
+if (!MILESTONE_NAME) {
+  throw new Error('You must specify a milestone name (eg: "2.15.0") as the second argument');
+}
+
+const WARNINGS = [
+  { labels: ['Breaking change'], title: 'Breaking changes' },
+  { labels: ['UI/UX'], title: 'UI/UX changes' },
+];
+
+const TYPES = [
+  { labels: ['Type: Feature'], title: 'Features' },
+  { labels: ['enhancement', 'Type: Improvement'], title: 'Improvements' },
+  { labels: ['Type: Security'], title: 'Security fixes' },
+  { labels: ['Type: Performance'], title: 'Performance improvements' },
+  { labels: ['bug', 'Type: Bug'], title: 'Bug fixes' },
+  { labels: ['Type: Technical issue'], title: 'Technical improvements' },
+];
+
+const PREFIXES_TO_IGNORE = [
+  'Type: Internal process',
+  'Won\'t fix:',
+  'Type: Investigation',
+];
+
+const getMilestone = async () => {
+  const response = await octokit.rest.issues.listMilestones({ owner: OWNER, repo: REPO_NAME });
+  return response.data.find(milestone => milestone.title === MILESTONE_NAME);
+};
+
+const getMilestoneNumber = async () => {
+  const milestone = await getMilestone();
+  if (!milestone) {
+    throw new Error(`Could not find milestone with the repo ${REPO_NAME} and name ${MILESTONE_NAME}`);
+  }
+  return milestone.number;
+};
+
+const validateIssue = issue => {
+  const matchingTypes = TYPES.filter(type => issue.labels.find(label => type.labels.includes(label.name)));
+  if (!matchingTypes.length) {
+    return `Issue doesn't have any Type label: ${issue.html_url}`;
+  }
+  if (matchingTypes.length > 1) {
+    return `Issue has too many Type labels: ${issue.html_url}`;
+  }
+};
+
+const validateIssues = issues => {
+  const errors = issues
+    .map(validateIssue)
+    .filter(error => !!error);
+  if (errors.length) {
+    console.error(JSON.stringify(errors, null, 2));
+    throw new Error('Some issues are in an invalid state');
+  }
+  return issues;
+};
+
+const getIssues = async (milestoneId) => {
+  const response = await octokit.rest.issues.listForRepo({
```

Comment:
I was going to complain about getting all the issues without filtering to only include "Closed" issues. I see the prior version of the script did the same, so it's not a regression, just more a question: Do we include everything on purpose?

Reply:
Yes we do. What it means is you can start drafting the release notes before all issues are closed. Obviously if you do this you need to update it if issues are added/removed but it's a useful starting point.